csm/common/timeseries.py

```diff
@@ -201,13 +201,13 @@ async def process_request(self, stats_id, panel, from_t, duration_t,
             duration_t = str(datetime.utcfromtimestamp(int(duration_t)).isoformat())+'.000Z'
             panel = panel.lower()
             metric_list, unit_list = await self._get_metric_list(panel, metric_list, unit)
-            res = await self._aggregate_metric(panel, from_t, duration_t,
-                                            interval, metric_list, query)
-            return await self._convert_payload(res, stats_id, panel, output_format, unit_list)
         except Exception as e:
             Log.debug("Failed to request stats %s" %e)
-            raise CsmInternalError("id: %s, Error: Failed to process timelion "
-                "request %s" %(stats_id,e))
+            raise InvalidRequest("id: %s, Error: Failed to process timelion "
```

Comment:
Catch both the exceptions InvalidRequest and Base exception. In case of Invalid request raise 400 otherwise 500.

Reply:
Error message should be generic. Do not explicitly use the names of 3rd party SW

Reply:
Addressed review comments